engine/engine/src/engine.cpp

```diff
@@ -1466,294 +1425,297 @@ namespace dmEngine
-                    int remainder = (int)((target_frametime - flip_dt) - engine->m_PreviousRenderTime);
-                    if (!engine->m_UseVariableDt && flip_dt < target_frametime && remainder > 1000) // only bother with sleep if diff b/w target and actual time is big enough
+                    uint32_t width = dmGraphics::GetWidth(engine->m_GraphicsContext);
+                    uint32_t height = dmGraphics::GetHeight(engine->m_GraphicsContext);
+                    uint32_t buffer_size = width * height * 4;
+
+                    dmGraphics::ReadPixels(engine->m_GraphicsContext, record_data->m_Buffer, buffer_size);
+
+                    dmRecord::Result r = dmRecord::RecordFrame(record_data->m_Recorder, record_data->m_Buffer, buffer_size, dmRecord::BUFFER_FORMAT_BGRA);
+                    if (r != dmRecord::RESULT_OK)
                     {
-                        DM_PROFILE(Engine, "SoftwareVsync");
-                        while (remainder > 500) // dont bother with less than 0.5ms
-                        {
-                            uint64_t t1 = dmTime::GetTime();
-                            dmTime::Sleep(100); // sleep in chunks of 0.1ms
-                            uint64_t t2 = dmTime::GetTime();
-                            remainder -= (t2-t1);
-                        }
+                        dmLogError("Error while recoding frame (%d)", r);
                     }
                 }
-                uint64_t flip_time_start = dmTime::GetTime();
+                record_data->m_FrameCount++;
+            }
+        }
+        dmProfile::Release(profile);
 
-                dmGraphics::Flip(engine->m_GraphicsContext);
+        ++engine->m_Stats.m_FrameCount;
+    }
 
-                engine->m_FlipTime = dmTime::GetTime();
-                engine->m_PreviousRenderTime = engine->m_FlipTime - flip_time_start;
+    static void CalcTimeStep(HEngine engine, float& step_dt, uint32_t& num_steps)
+    {
+        uint64_t time = dmTime::GetTime();
+        uint64_t frame_time = time - engine->m_PreviousFrameTime; // The actual time between two engine frames
+        engine->m_PreviousFrameTime = time;
 
-                RecordData* record_data = &engine->m_RecordData;
-                if (record_data->m_Recorder)
-                {
-                    if (record_data->m_FrameCount % record_data->m_FramePeriod == 0)
-                    {
-                        uint32_t width = dmGraphics::GetWidth(engine->m_GraphicsContext);
-                        uint32_t height = dmGraphics::GetHeight(engine->m_GraphicsContext);
-                        uint32_t buffer_size = width * height * 4;
+        float frame_dt = (float)(frame_time / 1000000.0);
 
-                        dmGraphics::ReadPixels(engine->m_GraphicsContext, record_data->m_Buffer, buffer_size);
+        // Never allow for large hitches
+        if (frame_dt > 0.5f) {
+            frame_dt = 0.5f;
+        }
```

Comment:
Good to know about this cap.
We might want is as a setting, but until anyone requests it...
SHould probably be documented though.